plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -202,6 +212,73 @@ private static boolean isAggregate(ExpressionContext expressionContext)
+            case FUNCTION:
+                if (isAggregate(expressionContext)) {
+                    return true;
+                }
+                for (ExpressionContext argument : expressionContext.getFunction().getArguments()) {
+                    if (hasAggregate(argument)) {
+                        return true;
+                    }
+                }
+                return false;
+        }
+        throw new PinotException(PINOT_EXCEPTION, Optional.empty(), format("Unsupported expression type '%s'", expressionContext.getType()));
+    }
+
+    // Extracted from org.apache.pinot.core.query.reduce.AggregationDataTableReducer
+    private static DataSchema getPreAggregationDataSchema(QueryContext queryContext)
+    {
+        AggregationFunction[] aggregationFunctions = queryContext.getAggregationFunctions();
+        int numAggregationFunctions = aggregationFunctions.length;
+        String[] columnNames = new String[numAggregationFunctions];
+        DataSchema.ColumnDataType[] columnDataTypes = new DataSchema.ColumnDataType[numAggregationFunctions];
+        for (int i = 0; i < numAggregationFunctions; i++) {
+            AggregationFunction aggregationFunction = aggregationFunctions[i];
+            columnNames[i] = aggregationFunction.getResultColumnName();
+            columnDataTypes[i] = aggregationFunction.getFinalResultColumnType();
+        }
+        return new DataSchema(columnNames, columnDataTypes);
+    }
+
+    private static Map<String, PinotColumnNameAndTrinoType> getAggregateTypes(SchemaTableName schemaTableName, QueryContext queryContext, Map<String, ColumnHandle> columnHandles)
+    {
+        // A mapping from pinot expression to the returned pinot column name and trino type
+        // Note: the column name is set by the postaggregation schema
+        List<ExpressionContext> aggregateColumnExpressions = queryContext.getSelectExpressions().stream()
+                .filter(DynamicTableBuilder::hasAggregate)
+                .collect(toImmutableList());
+        queryContext = new QueryContext.Builder()
+                .setAliasList(queryContext.getAliasList())
+                .setSelectExpressions(aggregateColumnExpressions)
+                .build();
+        DataSchema preAggregationSchema = getPreAggregationDataSchema(queryContext);
+        PostAggregationHandler postAggregationHandler = new PostAggregationHandler(queryContext, preAggregationSchema);
+        DataSchema postAggregtionSchema = postAggregationHandler.getResultDataSchema();
+        ImmutableMap.Builder<String, PinotColumnNameAndTrinoType> aggregationTypesBuilder = ImmutableMap.builder();
+        for (int index = 0; index < postAggregtionSchema.size(); index++) {
+            aggregationTypesBuilder.put(rewriteExpression(schemaTableName, aggregateColumnExpressions.get(index), columnHandles).toString(), new PinotColumnNameAndTrinoType(postAggregtionSchema.getColumnName(index), toTrinoType(postAggregtionSchema.getColumnDataType(index))));
```

Comment:
split arguments into new lines, more readable

Reply:
What about column references though? Do they not require quoting?

Reply:
I left a comment in the code: the identifiers are purposely not quoted as this matches the returned column from ResultTable. I added a unit test with `percentile` since this was one of the cases which required this fix - this matches the code in the AggregationDataTableReducer which is called by the PostAggregationHandler.

Reply:
I left a comment in the code: the identifiers are purposely not quoted as this matches the returned column from ResultTable. I added a unit test with `percentile` since this was one of the cases which required this fix - this matches the code in the AggregationDataTableReducer which is called by the PostAggregationHandler.